### DISCUSSION TITLE
Add support for Immersive Translate to translate local files offline

### DISCUSSION THREAD

**Discussion by vtkspider** [immersive-translate/immersive-translate#2098] [Ideas]
### 新功能 | New Feature

1. enable local file translation based on local LLM (ollama etc.)

### 为什么您想要添加这些功能 | Why you want to add these features

1. need to use immersive translate in security Intranet 

### 提议 | Proposal

Maybe it works, as just not connect to immersive-translate.com while translate local file

**Comment by theowenyoung** [comment]
yes, you can use custom api to translate local file now.

**Reply by vtkspider** [reply]
when translate pdf, it jumps to "https://app.immersivetranslate.com/pdf/?file=emptyfile". So it does not work offline. Is there any workaround?

**Reply by theowenyoung** [reply]
you need to enable file permission at Immersive Translate Extension Detail Settings on Browser.

**Reply by theowenyoung** [reply]
docs: https://documentation.sas.com/doc/en/gracclug/1.0/n1cv2euhlxw6zbn1r4xuhj8bbly1.htm

**Reply by vtkspider** [reply]
Many thanks!
Yes I did enable "Allow access to file URLs" in "Manage extensions". But when clicks pop-up translate, immersive translator add-on tries to access  "https://app.immersivetranslate.com/pdf/?file=emptyfile" to open and parse pdf file. So I get situation as showed in attached picture.
![微信图片_20240926160249](https://github.com/user-attachments/assets/2172a2b3-0cc9-4be6-9f23-9333b0fec1fb)

**Reply by theowenyoung** [reply]
it looks like you're using linux chrome?

it works fine on my macbook chrome.

https://github.com/user-attachments/assets/8080abc2-f9e7-40aa-884f-48616b26938b

**Reply by vtkspider** [reply]
So kindly reply me!
Maybe there is some misunderstand. I want to translate pdf without connect to internet. In your video the network is open. Maybe closing internet makes difference. 
In https://github.com/immersive-translate/immersive-translate/blob/main/dist/chrome/pdf/index.html, there is :
```
  <body style="padding:0px;margin:0px;">
    <iframe
      src="https://app.immersivetranslate.com/pdf/?file=emptyfile"
      width="100%"
      style="min-height: 100vh;border-width: 0px;"
    ></iframe>
```
   So no matter in macbook chrome or linux chrome, translating pdf only works after the pdf file is sent to app.immersivetranslate.com and parsed by pdf.js.
   
   Please help if thers is any workaround. Maybe pdf.js server on local machine?
